proto-public/pbmesh/v2beta1/pbproxystate/header_mutations.proto

```diff
@@ -46,4 +46,5 @@ enum AppendAction {
   // buf:lint:ignore ENUM_ZERO_VALUE_SUFFIX
   APPEND_ACTION_APPEND_IF_EXISTS_OR_ADD = 0;
   APPEND_ACTION_OVERWRITE_IF_EXISTS_OR_ADD = 1;
+  APPEND_ACTION_ADD_IF_ABSENT = 2;
```

Comment:
~~ Non-blocking and not a super strong opinion: Given we aren't committed to backwards compatibility yet, it might help avoid a future mismatch w/ Envoy to swap these two values so they match Envoy's cardinals, assuming we expect the enum sets to match. We did an internal enum swap previously here: https://github.com/hashicorp/consul/pull/18612